scripts/fixtures/door.gd

```diff
@@ -24,6 +24,9 @@ func toggle() -> void:
 		$Sprite.frame = 0
 	
 	vertical = not vertical
+	
+func on_queue_free() -> void:
```

Comment:
classic use case for an interface, IFreeable or IDisposable. Unfortunately Godot doesn't support them. [There is an addon that can mimic the behavior using traits](https://github.com/Earewien/godot-traits), but it's a little bloaty IMO.